### DISCUSSION TITLE
Removing parameters for "customize y axis range", or adding a greater range of options

### DISCUSSION THREAD

**Discussion by bkuz114** [NightscoutFoundation/xDrip#3479] [Ideas]
Hi folks. First post. First off, really appreciative of xDrip+. So much better than using libre3 app. My ask in this thread is likely a minority use case, so feel free to disregard, but going to put it out there anyway...

For context, the setting I'm referring to is in settings --> xDrip+ Display Settings -->Graph settings --> Customize y axis range. This setting allows you to change the minimum and maxiumum y value for the main graph.

Currently, if you click "Adjust y axis maximum" you are given only 5 options: 210 mg/dl, 230 mg/dl, 250 mg/dl, 300 mg/dl, and 400 mg/dl.

**My idea**: is it possible just input the maximum value you want, rather than having to select from these static values? Or have perhaps an additional field where you can put in a custom max value, if you don't like any of those values? Or, add a greater range of options?

**Use case:** I realize those y-values are very useful for people with diabetes, and so you might think "what is the need of this?". In my case, I do not have diabetes. Rather, I have some metabolic irregularities that result in chronic episodes of hypoglyemia (hence need of CGM). My blood sugar is rarely over 100 mg/dl, and is low (sub 70 mg/dl) multiple times per day. Currently when I use xDrip+, the lowest I can set the "max" for the y-axis is is 210 mg/dl. Consequently, the graph is mostly empty space, as my blood glucose stays in a much narrower, and much lower range. It would be nice to be able to customize the y axis in such a way that I can see my glucose readings more clearly (Rather than mostly empty space).

**One quick solution idea?**: Rather than making the value customizable, simply adding 150 mg/dl as an option would be incredibly useful. I wonder if this would be something easy to implement?

**Comment by Navid200** [comment]
There are times a diabetic forgets to take insulin.  We all make plans.  But, life sometimes does not go exactly as planned.  When one ends up having readings that are greater than their chosen max y axis, xDrip automatically extends the y axis and ignores the set customization so that all readings within the last 24 hours can be visible on screen.

A consequence of this is that if you one day forget insulin and your readings temporarily go up, the slopes on your screen will decrease for 24 hours.  This is not good.
That is the reason we did not want the max to be able to be reduced too much.
By the way, we have two open issues related to this:
https://github.com/NightscoutFoundation/xDrip/issues/2173
https://github.com/NightscoutFoundation/xDrip/issues/3081

Considering all of that, just adding a new lower option for the max value without changing anything else is not a possibility.

Just to put all of this in context, please note that Dexcom app only gives you two options: 400 or 300.
The max value for xDrip used to be only 250.
We have offered more flexibility recently by adding the options.  

There has been a suggestion that we always use the set max and min to calculate a range and always force the range and move the entire range up and or down such that the values for the past 4 hours always remain visible even if the current readings may be greater than the set max.  This way, the slopes never change.  
But, this should still allow the user to scroll so that they can see the readings during the past 24 hours that may be outside the current visible range if they choose to.  

This is a complicated solution that I have wanted to implement.  However, I have been busy with other issues so far.
This may also allow us to offer a lower optional range because it addresses occasional high readings without changing the slopes.

@trallen may be able to help us with this.

**Comment by bkuz114** [comment]
Thanks for the response. It sounds like a lot of thought went into this design decision, and that it makes the app safer for the typical user (at least the way things are currently implemented). I would not want to suggest a change that could put anyone at risk, of course, so I detract my idea then. I think I am likely in the less than 1% of users of the CGM, so I don't want to be the reason a complicated project is implemented or anything (especially when my reason for posing the idea in the first place is so minor, and honestly not that big of a deal). Keeping people safe is the top thing I would hope for. So thanks for explaining the reasons.

By the way - what you said about how dexcom only gives options of 400 and 300. I am with you on this. The reason I am using this xDrip app is because it seems the stock apps for some of these devices have very little flexibility. For example (and the main reason I found and am using xDrip+), in libre 3 app, for a low glucose alarm, you can't set it any lower than 60 mg/dl. I need an alarm that will alert me at a lower level when I'm asleep, and it's the main reason I need a CGM (To wake me up when it gets too low at night). I am super appreciative that you guys don't have those thresholds, and that you give this flexibility. This is life-changing for me to be honest, to no longer have to be worried about sleeping at night. So I hope my idea didn't come across as an insult towards the app.

**Reply by PhilTU** [reply]
I do agree with you about wasted screen real estate. For users that are using CGMs for metabolic conditions other than diabetes anything over 200 is probably just "high" and the value is not of interest, if it occurs at all. But 210 is not so bad compared to Librelink's 380.

**Reply by bkuz114** [reply]
Agreed completely regarding librelink; the y-axis is not customizable, and their entire graph is functionally useless for me, as it mostly appears as a flat line as a result. Whenever I want to visualize my blood sugar graph, I open xDrip+.

Again, this app is made primarily for people with diabetes, and it seems there's very calculated reasons why the graph is designed the way it is. I'm completely OK with it, and I believe it still serves my needs, as I can still see the ups and downs. If there's some wasted space - no biggie. My biggest concern is having low blood sugar alarms which it provides me with flawlessly. I've been able to be woken 4 times this week for dangerous lows. First time in many years that I have been able to sleep well during these issues, because I don't have to just keep waking up frantically every 30 minutes to see if I'm OK. Really incredible stuff, and a well-done app on top of it. Great UI, and I have not yet experienced bugs. Well done, xDrip developers!
